text/3389-manifest-lint.md

```diff
@@ -0,0 +1,250 @@
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+A concern brought up in
+[rust-lang/rust-clippy#1313](https://github.com/rust-lang/rust-clippy/issues/1313)
+was that this will pass lints unconditionally to the underlying tool, leading
+to "undefined lint" warnings when used on earlier versions, requiring that
+warning to also be suppressed, reducing its value.  However, in the "Future
+possibility's", we mention direct support for tying lints to rust versions.
+
+This does not allow sharing lints across workspaces.
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+This could be left to `clippy.toml` but that leaves `rustc` without a solution.
+
+`[lints]` could be `[package.lints]`, tying it to the package unlike `[patch]`
+and other fields that are more workspace related.  Instead, we used
+`[dependencies]` as our model.
+
+`[lints]` could be `[lint]` but we decided to follow the precedence of `[dependencies]`.
+
+Instead of using `::` as a separator between tool and lint (e.g.
+`clippy::enum_glob_use`), we could use TOML dotted keys for this (e.g.
+`clippy.enum_glob_use`).  This has the advantage of allowing unquoted keys at
+the cost of not being able to copy/paste the lint name from the tool's output
+into the fileV
+
+We could support platform or feature specific settings, like with
+`[lints.<target>]` or `[target.<target>.lints]` but
+- There isn't a defined use case for this yet besides having support for `cfg(feature = "clippy")` or
+  which does not seem high enough priority to design
+  around.
+- `[lints.<target>]` runs into ambiguity issues around what is a `<target>`
+  entry vs a `<lint>` entry in the `[lints]` table.
+- We have not yet defined semantics for sharing something like this across a
+  workspace
+
+Instead of using workspace inheritance for `[lint]`, we could make it
+workspace-level configuration, like `[patch]` which is automatically applied to
+all workspace members.  However, `[patch]` and friends are because they affect
+the resolver / `Cargo.toml` and so they can only operate at the workspace
+level.  `[lints]` is more like `[dependencies]` in being something that applies
+at the package level but we want shared across workspaces.
+
+Instead of traditional workspace inheritance where there is a single value to
+inherit with `workspace = true`, we could have `[workspace.lints.<preset>]`
+which defines presets and the user could do `lints.<preset> = true`.  The user
+could then name them as they wish to avoid collision with rustc lints.
+
+Instead of the `[lints]` table being `lint = "level"`, we could organize
+it around `level = ["lint", ...]` like some other linters do (like
+[ruff](https://beta.ruff.rs/docs/configuration/)) but this works better for
+logically organizing lints, highlighting what changed in diffs, and for
+possibly adding lint-specific configuration in the future.
+
+# Prior art
+[prior-art]: #prior-art
```

Comment:
There is also [`cargo cranky`](https://github.com/ericseppanen/cargo-cranky) that is worth mentioning here. This also uses the `level = [ ... ]` syntax.

Reply:
Good call; now included